internal/concurrent/range.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package concurrent
+
+import (
+	"log"
+	"reflect"
+	"sync"
+
+	"github.com/thriftrw/thriftrw-go/internal"
+)
+
+var (
+	_typeOfError = reflect.TypeOf((*error)(nil)).Elem()
+	_typeOfInt   = reflect.TypeOf(int(0))
+)
+
+// Range calls the function f on all items in c concurrently and waits for all
+// calls to finish.
+//
+// c may be a slice or a map. If c is a map, f must accept the key and the value
+// as its arguments, and otherwise it must accept an int index and the value as
+// its arguments.
+//
+// f may return nothing or error.
+func Range(c, f interface{}) error {
```

Comment:
Suggest renaming `f` to `fn` or `func` - I think `f` is abbreviated too aggressively.